File: 02358.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
	int n; cin >> n;

	vector<int> x, y;
	for (int i = 0; i < n; i++) {
		int a, b; cin >> a >> b;
		x.push_back(a);
		y.push_back(b);
	}

	int line = 0;

	// x 좌표 계산
	sort(x.begin(), x.end());
	int cnt = 1, num = x[0];
	for (int i = 1; i < x.size(); i++) {
		if (x[i] == num) cnt++;
		else {
			line += (cnt >= 2);
			cnt = 1;
			num = x[i];
		}
	}
	line += (cnt >= 2);

	// y 좌표 계산
	sort(y.begin(), y.end());
	cnt = 1, num = y[0];
	for (int i = 1; i < y.size(); i++) {
		if (y[i] == num) cnt++;
		else {
			line += (cnt >= 2);
			cnt = 1;
			num = y[i];
		}
	}
	line += (cnt >= 2);

	cout << line << endl;
}

File: 02358.py
xy = []
for i in range(int(input())):
    xy.append(list(map(int, input().split())))

line = 0

x = sorted(xy, key=lambda w: w[0])
cnt = 1; num = x[0][0]
for i in range(1, len(x)):
    if x[i][0] == num:
        cnt += 1
    else:
        line += (cnt >= 2)
        cnt = 1
        num = x[i][0]
line += (cnt >= 2)

y = sorted(xy, key=lambda w: w[1])
cnt = 1; num = y[0][1]
for i in range(1, len(y)):
    if y[i][1] == num:
        cnt += 1
    else:
        line += (cnt >= 2)
        cnt = 1
        num = y[i][1]
line += (cnt >= 2)

print(line)
